docker-compose.desktop.yml

```diff
@@ -0,0 +1,77 @@
+# Example compose file for a running a local Firezone instance on
+# macOS or Windows.
+#
+# Note: This file is meant to serve as a template. Please modify it
+# according to your needs. Read more about Docker Compose:
+#
+# https://docs.docker.com/compose/compose-file/
+#
+#
+x-deploy: &default-deploy
+  restart_policy:
+    condition: on-failure
+    delay: 5s
+    max_attempts: 3
+    window: 120s
+  update_config:
+    order: start-first
+
+version: '3.7'
+
+services:
+  caddy:
+    image: caddy:2
+    volumes:
+      - ${FZ_INSTALL_DIR:-.}/caddy:/data/caddy
+    ports:
+      - 80:80
+      - 443:443
+        # See Caddy's documentation for customizing this line
+        # https://caddyserver.com/docs/quick-starts/reverse-proxy
+    command: caddy reverse-proxy ${CADDY_OPTS:-} --to firezone:13000 --from ${EXTERNAL_URL:?err}
```

Comment:
Does the command line work with `*` like:
```
command: caddy reverse-proxy ${CADDY_OPTS:-} --to firezone:13000 --from *
```

So that it works with ip?

Reply:
Unfortunately that doesn't do the trick:

```
firezone-caddy-1  | {"level":"debug","ts":1667524342.2751524,"logger":"http.stdlib","msg":"http: TLS handshake error from 172.26.0.1:63014: no certificate available for '172.26.0.3'"}
```

Perhaps hardcoding the container IP and using that will work.

Reply:
hmm I'd had assumed that caddy picked the certificate's domain based on the site's domain

If I understand the docs correctly [here](https://caddyserver.com/docs/automatic-https#hostname-requirements) it seems like a single `*` should work.

Reply:
I did some testing and if you remove the `--from` part of the command completely, so it is:

```
caddy reverse-proxy ${CADDY_OPTS:-} --to firezone:13000
```

if the user uses 127.0.0.1 as the external url, it does work but only when the user doesn't use it in the browser, meaning they have to type `localhost`.

But maybe with that `tls selfsigned` it works. Although the documentation now states that it will automatically serve ips with self-signed certs now [here](https://caddyserver.com/docs/automatic-https#overview)

Reply:
Yeah, currently testing out a Caddyfile that may do the trick.

Reply:
Fixed.

Reply:
<img width="1552" alt="Screen Shot 2022-11-03 at 7 18 52 PM" src="https://user-images.githubusercontent.com/167144/199871111-dafd377d-d9b7-4fad-909a-81cea22e4d3d.png">